tests/MongoDB.Driver.Core.Tests/Core/Connections/ConnectionDescriptionTests.cs

```diff
@@ -30,13 +31,36 @@ public class ConnectionDescriptionTests
             "{ ok: 1, version: \"2.6.3\" }"
         ));
 
+        private static readonly IEnumerable<CompressorType> __compression = new[] { CompressorType.Zlib };
+
         private static readonly ConnectionId __connectionId = new ConnectionId(
             new ServerId(new ClusterId(), new DnsEndPoint("localhost", 27017)));
 
         private static readonly IsMasterResult __isMasterResult = new IsMasterResult(BsonDocument.Parse(
+            "{ ok: 1, maxWriteBatchSize: 10, maxBsonObjectSize: 20, maxMessageSizeBytes: 30, compression: ['zlib'] }"
+        ));
+
+        private static readonly IsMasterResult __isMasterResultWithoutCompression = new IsMasterResult(BsonDocument.Parse(
             "{ ok: 1, maxWriteBatchSize: 10, maxBsonObjectSize: 20, maxMessageSizeBytes: 30 }"
         ));
 
+        [Fact]
+        public void Compression_should_not_be_null()
+        {
+            var subject = new ConnectionDescription(__connectionId, __isMasterResultWithoutCompression, __buildInfoResult);
+
+            subject.AvailableCompressors.Count.Should().Be(0);
+        }
+
+        [Fact]
+        public void Compression_should_return_Compression()
```

Comment:
A better name is probably

    AvailableCompressors_should_return_expected_result